### PR TITLE
Updated oi

### PR DESCRIPTION
**Pull request by rajitzg**
## Description

Cleaned up OI and consolidated current mechanisms/controls.

Added input constants for OI
Added OIControls.md, detailing driver and box op controls
Added procedure for reversing gyro intake for autonomous
Added a gyro reset at the start of autonomous, which is set to 180deg because the robot will face the drivers at the start of a match.
Added fine tuning controls. If a set button is pressed, swerve drive controls will be lowered so the robot can be finely adjusted.

## How Has This Been Tested?

- [ ] Unit tests:
- [ ] Simulator testing:
- [x] On-robot bench testing: Tested addition of fine tuning and shuffleboard pressure
- [x] On-robot field testing: Tested all changes on carpet field

### CODE REVIEW

File: src/main/java/com/team766/robot/OI.java [hunk 1]
```diff
@@ -52,155 +51,62 @@ public void run(Context context) {
...
-			if (joystick0.getButton(inButton)) {
+			// Sets intake state based on held down button
+			// Idles intake if either button released
+			if (controlPanel.getButtonPressed(InputConstants.INTAKE)) {
 				Robot.intake.startIntake();
 				Robot.storage.beltIn();
-			} else if (joystick0.getButton(outButton)) {
+			} else if (controlPanel.getButtonPressed(InputConstants.OUTTAKE)) {
 				Robot.intake.reverseIntake();
 				Robot.storage.beltOut();
```
Comment (jamesi8086) [inline src/main/java/com/team766/robot/OI.java]: recall you mentioned during outtake the intake pistons would remain up, not sure if its reflected here

File: src/main/java/com/team766/robot/OI.java [hunk 2]
```diff
@@ -52,155 +51,62 @@ public void run(Context context) {
...
+			// Moves the robot if there are joystick inputs
+			if (Math.abs(leftJoystickX) + Math.abs(leftJoystickY) +  Math.abs(rightJoystickX) > 0) {
+				context.takeOwnership(Robot.drive);
+				// If a button is pressed, drive is just fine adjustment
+				if (leftJoystick.getButton(InputConstants.FINE_DRIVING)) {
+					Robot.drive.swerveDrive((leftJoystickX * FINE_DRIVING_COEFFICIENT), (-leftJoystickY * FINE_DRIVING_COEFFICIENT), (rightJoystickX * FINE_DRIVING_COEFFICIENT));
+				} else {
+					Robot.drive.swerveDrive((leftJoystickX), (-leftJoystickY), (rightJoystickX));
+				}
+			} else {
```
Comment (jamesi8086) [inline src/main/java/com/team766/robot/OI.java:156]: do you want to add a "slow mode" or "precision mode" for the driver to carefully get onto the charging station? can  be in another PR

**Review by jamesi8086 — APPROVED** [review]
LGTM with comments

**Comment by BCNOFNeNaMg** [comment]
LGTM
